deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/Hdf5Archive.java

```diff
@@ -61,6 +61,21 @@ public Hdf5Archive(String archiveFilename) {
         this.file = new hdf5.H5File(archiveFilename, H5F_ACC_RDONLY);
     }
 
+    private hdf5.Group[] openGroups(String ... groups) {
+        hdf5.Group[] groupArray = new hdf5.Group[groups.length];
+        groupArray[0] = this.file.asCommonFG().openGroup(groups[0]);
+        for (int i = 1; i < groups.length; i++) {
+            groupArray[i] = groupArray[i - 1].asCommonFG().openGroup(groups[i]);
+        }
+        return groupArray;
+    }
+
+    private void closeGroups(hdf5.Group[] groupArray) {
```

Comment:
What are your thoughts on 
```
hdf5.Group thisGroup = groupArray[i];
thisGroup._close();
```
?

Reply:
Examples from the HDF5 documentation uses the `delete` operator and that's called with `dellocate()`...

Reply:
I was inspired by §5.4 here : https://support.hdfgroup.org/HDF5/doc1.6/UG/09_Groups.html

Reply:
That refers to the C API. The C++ API seems to be a bit wonky...

Reply:
Haha, fair enough.